apps/OboeTester/app/src/main/java/com/mobileer/oboetester/RoundTripLatencyActivity.java

```diff
@@ -195,15 +216,28 @@ public String getLastReport() {
         private Runnable runnableCode = new Runnable() {
             @Override
             public void run() {
-                String message;
+                double timestampLatency = -1.0;
+                int state = getAnalyzerState();
+                if (state == STATE_MEASURE_BACKGROUND || state == STATE_IN_PULSE) {
+                    timestampLatency = measureTimestampLatency();
+                    if (timestampLatency > 0) {
+                        mTimestampLatencySum += timestampLatency;
+                        mTimestampLatencyCount++;
+                    }
+                }
 
+                String message;
                 if (isAnalyzerDone()) {
-                    message = mLatencyAverager.onAnalyserDone();
-                    message += onAnalyzerDone();
+                    if (mLatencyAverager.isActive()) {
+                        message = mLatencyAverager.onAnalyserDone();
+                    } else {
+                        message = getResultString();
+                    }
+                    onAnalyzerDone();
                 } else {
                     message = getProgressText();
                     message += "please wait... " + counter + "\n";
-                    message += convertStateToString(getAnalyzerState());
+                    message += convertStateToString(getAnalyzerState()) + "\n";
 
                     // Repeat this runnable code block again.
                     mHandler.postDelayed(runnableCode, SNIFFER_UPDATE_PERIOD_MSEC);
```

Comment:
Wouldn't having this run periodically skew the results towards results that have higher latency compared to if we collected every frame?

Reply:
> collected every frame

At what frame rate?

Timestamps are only collected a few times per second. We are probably already oversampling.

Reply:
Is there a way to sample at the same frequency as the device?

Oversampling is fine. If we know we may be oversampling, we should sample even more frequently. See Nyquist's theorem

Reply:
Over-sampling a function that does not change does not give you more data.
I tested and we are under-sampling. That is fine. We just want to measure the latency several time over a few seconds to see how much variation there. it does not vary much.